tests/testImageDecorrelation.py

```diff
@@ -0,0 +1,253 @@
+from __future__ import absolute_import, division, print_function
+#
+# LSST Data Management System
+# Copyright 2008-2016 AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+
+import unittest
+import numpy as np
+
+import lsst.utils.tests
+import lsst.afw.image as afwImage
+import lsst.afw.geom as afwGeom
+import lsst.afw.math as afwMath
+import lsst.meas.algorithms as measAlg
+# import lsst.ip.diffim as ipDiffim
+# import lsst.daf.base as dafBase
+import lsst.pex.logging as pexLog
+
+from lsst.ip.diffim.imageDecorrelation import performExposureDecorrelation as doDecorr
+
+def singleGaussian2d(x, y, xc, yc, sigma_x=1., sigma_y=1., theta=0.):
+    """! Generate a 2-d Gaussian, possibly elongated and rotated, on a grid of pixel
+    coordinates given by x,y.
+    @param x,y each a 1-d numpy.array containing x- and y- coordinates for independent variables.
+    @param xc,yc each a float giving the centroid of the gaussian
+    @param sigma_x,sigma_y each a float giving the sigma of the gaussian
+    @param theta a float giving the rotation of the gaussian (degrees)
+    @return a 2-d numpy.array containing the normalized 2-d Gaussian
+    """
+    theta = (theta/180.) * np.pi
+    cos_theta2, sin_theta2 = np.cos(theta)**2., np.sin(theta)**2.
+    sigma_x2, sigma_y2 = sigma_x**2., sigma_y**2.
+    a = cos_theta2/(2.*sigma_x2) + sin_theta2/(2.*sigma_y2)
+    b = -(np.sin(2.*theta))/(4.*sigma_x2) + (np.sin(2.*theta))/(4.*sigma_y2)
+    c = sin_theta2/(2.*sigma_x2) + cos_theta2/(2.*sigma_y2)
+    xxc, yyc = x-xc, y-yc
+    out = np.exp(-(a*(xxc**2.) + 2.*b*xxc*yyc + c*(yyc**2.)))
+    out /= out.sum()
+    return out
+
+
+def makeFakeImages(xim=None, yim=None, sig1=0.2, sig2=0.2, psf1=2.2, psf2=3.3, offset=None,
+                   psf_yvary_factor=0., varSourceChange=1/50., theta1=0., theta2=0., im2background=0.,
+                   n_sources=500, seed=66, verbose=False):
+    """! Make the two "images". im1 is the template, im2 is the science image.
```

Comment:
I think it might be clearer to say: `Make two images: a template and a science image"